src/gmp.c

```diff
@@ -18527,6 +18577,157 @@ handle_modify_config (gmp_parser_t *gmp_parser, GError **error)
+  result->description = create_report_data->result_description;
+  // sometimes host has newlines in it, so we 0 terminate first newline
+  // According to
+  // https://www.freebsd.org/cgi/man.cgi?query=strcspn&sektion=3
+  // strcspn returns the number of chars spanned so it should be safe
+  // without double checking.
+  create_report_data
+    ->result_host[strcspn (create_report_data->result_host, "\n")] = 0;
+  result->host = create_report_data->result_host;
+  result->hostname = create_report_data->result_hostname;
+  result->nvt_oid = create_report_data->result_nvt_oid;
+  result->scan_nvt_version = create_report_data->result_scan_nvt_version;
+  result->port = create_report_data->result_port;
+  result->qod = create_report_data->result_qod;
+  result->qod_type = create_report_data->result_qod_type;
+  result->severity = create_report_data->result_severity;
+  result->threat = create_report_data->result_threat;
+  if (result->host)
+    {
+      for (unsigned int i = 0; i < create_report_data->result_detection->len;
+           i++)
+        {
+          // prepare detection to be found within
+          // result_detection_reference
+          detection_detail_t *detection =
+            (detection_detail_t *) g_ptr_array_index (
+              create_report_data->result_detection, i);
+
+          host_detail_t *detail;
+          // used to find location within report_host_details via
+          // - oid as source_name
+          // - detected_at as name
+          // could be simplified by using detected_by as a key for
+          // location,
+          // but may have unwanted sideeffects.
+          detail = g_malloc (sizeof (host_detail_t));
+          detail->ip = g_strdup (result->host);
+          detail->name = g_strdup ("detected_at");
+          detail->source_desc = g_strdup ("create_report_import");
+          detail->source_name = g_strdup (
+            detection->source_oid); // verify when detected_at || detected_by
+          detail->source_type = g_strdup ("create_report_import");
+          detail->value = g_strdup (detection->location);
+          array_add (create_report_data->details, detail);
+          // used to find oid within report_host_details via
+          // - oid as source_name
+          // - detected_by as name
+          // could be simplified by using detected_at as a key for
+          // oid,
+          // but may have unwanted sideeffects.
+          // Doubling the  data set.
+          detail = g_malloc (sizeof (host_detail_t));
+          detail->ip = g_strdup (result->host);
+          detail->name = g_strdup ("detected_by");
+          detail->source_desc = g_strdup ("create_report_import");
+          detail->source_name = g_strdup (detection->source_oid);
+          detail->source_type = g_strdup ("create_report_import");
+          detail->value = g_strdup (detection->source_oid);
+          array_add (create_report_data->details, detail);
```

Comment:
Besides the basic `detected_by` details, there are also `detected_by@[...]` details that should be created here.
These also contain the location (as the `[...]` in the name) in case different versions of a vulnerable product were found at different locations on the same host (like multiple ports or installation paths).